.github/workflows/coverage.yml

```diff
@@ -50,11 +50,28 @@ jobs:
       with:
         tool: cargo-llvm-cov@0.6.19,cargo-nextest@0.9.104
     - name: Generate coverage report
-      run: cargo llvm-cov nextest --workspace --lcov --output-path lcov.info --partition hash:${{ matrix.partition }}/8 --run-ignored all
+      run: cargo llvm-cov nextest --workspace --lcov --output-path lcov-${{ matrix.partition }}.info --partition hash:${{ matrix.partition }}/8 --run-ignored all
+    - name: Upload coverage artifact
+      uses: actions/upload-artifact@330a01c490aca151604b8cf639adc76d48f6c5d4 # v5.0.0
+      with:
+        name: coverage-lcov-${{ matrix.partition }}
+        path: lcov-${{ matrix.partition }}.info
+
+  UploadCoverage:
+    name: Upload Combined Coverage
+    runs-on: ubuntu-latest
+    needs: Coverage
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@08eba0b27e820071cde6df949e0beb9ba4906955 # v4
+    - name: Download coverage reports
```

Comment:
should we delete after this?

Reply:
Shouldn't need to. There's no official "delete artifact" action from GH, and github will automatically clean them up after the retention period.

Went ahead and specified the retention period as 1d (minimum), in-case our repo's settings are higher for other artifacts.